templates/controller.js

```diff
@@ -5,116 +5,154 @@ const {modelName} = require({modelPath});
+            return res.status(404).json({
+                message: 'No such {name}'
+            });
+        }
+
+        return res.json({name});
+    }
+    catch (err) {
+        return res.status(500).json({
+            message: 'Error when getting {name}.',
+            error: err
+        });
+    }
+
+};
+
+/**
+ * {controllerName}.create()
+ */
+const create = async (req, res) => {
+
+    let {name} = new {modelName}({{createFields}});
+
+    try {
+        {name} = await {name}.save();
+        return res.status(201).json({name});
+    }
+    catch (err) {
+        return res.status(500).json({
+            message: 'Error when creating {name}.',
+            error: err
         });
-    },
+    }
+}
+
+/**
+* {controllerName}.update()
+*/
 
-    /**
-     * {controllerName}.create()
-     */
-    create: function (req, res) {
-        const {name} = new {modelName}({{createFields}
+const update = async (req, res) => {
+    let id = req.params.id;
+    let {name};
+
+    try {
+        {name} = await {modelName}.findOne({_id: id}).exec();
+    }
+    catch (err) {
+        return res.status(500).json({
+            message: 'Error when getting {name}',
+            error: err
         });
+    }
 
-        {name}.save(function (err, {name}) {
-            if (err) {
-                return res.status(500).json({
-                    message: 'Error when creating {name}',
-                    error: err
-                });
-            }
 
-            return res.status(201).json({name});
+    {updateFields}
+
+    try {
+        {name} = await {name}.save();
+        return res.json({name});
+    }
+    catch (err) {
+        return res.status(500).json({
+            message: 'Error when updating {name}.',
+            error: err
         });
-    },
-
-    /**
-     * {controllerName}.update()
-     */
-    update: function (req, res) {
-        const id = req.params.id;
-
-        {modelName}.findOne({_id: id}, function (err, {name}) {
-            if (err) {
-                return res.status(500).json({
-                    message: 'Error when getting {name}',
-                    error: err
-                });
-            }
-
-            if (!{name}) {
-                return res.status(404).json({
-                    message: 'No such {name}'
-                });
-            }
-
-            {updateFields}
-            {name}.save(function (err, {name}) {
-                if (err) {
-                    return res.status(500).json({
-                        message: 'Error when updating {name}.',
-                        error: err
-                    });
-                }
-
-                return res.json({name});
-            });
+    }
+
+};
+
+
+/**
+ * {controllerName}.remove()
+ */
+const remove = async (req, res) => {
+    let id = req.params.id;
+
+    try {
+        let {name} = await {modelName}.findByIdAndRemove(id).exec();
+        return res.status(204).json();
+    }
+    catch (err) {
```

Comment:
lint : to put on a line.
`} catch (err) {`

Reply:
Done